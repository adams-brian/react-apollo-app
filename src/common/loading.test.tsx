import { mount } from 'enzyme';
import * as React from 'react';
import * as sinon from 'sinon';

import Loading from './loading';

describe("Loading", () => {

  let clock: sinon.SinonFakeTimers;
  const element = <Loading />;

  beforeEach(() => {
    clock = sinon.useFakeTimers();
  });

  afterEach(() => {
    clock.restore();
  });

  it('should render null for 500ms, then loading', () => {
    const component = mount(element);
    expect(component.html()).toMatchSnapshot();
    clock.tick(500);
    expect(component.html()).toMatchSnapshot();
    clock.tick(5000);
    expect(component.html()).toMatchSnapshot();
  });

  it('should clear the timeout on unmount', () => {
    const component = mount(element);
    expect(component.html()).toMatchSnapshot();
    component.unmount();
    clock.tick(5000);
    expect(component.html()).toMatchSnapshot();
  });

});
