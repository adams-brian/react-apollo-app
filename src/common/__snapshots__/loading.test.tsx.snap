// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should clear the timeout on unmount 1`] = `null`;

exports[`Loading should clear the timeout on unmount 2`] = `null`;

exports[`Loading should render null for 500ms, then loading 1`] = `null`;

exports[`Loading should render null for 500ms, then loading 2`] = `"<div>Loading...</div>"`;

exports[`Loading should render null for 500ms, then loading 3`] = `"<div>Loading...</div>"`;
